salt/modules/virt.py

```diff
@@ -1531,6 +1565,10 @@ def init(name,
         _get_domain(conn, name).create()
     conn.close()
 
+    # Cleanup the temporary directory needed for initrd and kernel (if
+    # downloaded)
+    shutil.rmtree(temp_dir)
+
```

Comment:
Huh... you shouldn't remove it too early. I have no idea at what time we can remove it safely: the VM needs to have loaded the initrd at least. At least in that part of the code the VM may not be started at all: `conn.defineXML(vm_xml)` is only defining it without starting it.

In the end may be the best thing to do is to let the user delete those files when he doesn't need them anymore. After all the path is in the XML description and we can't know how many times the machine needs to be booted with those external kernel/initrd.

To ensure the files are in sync I think we would recommend the user to combine `virt.running` with local path and `file.managed` states caching the initrd and kernel before running the VM. But this is out of the scope of your PR.